st2common/st2common/persistence/trigger.py

```diff
@@ -84,3 +84,7 @@ class TriggerInstance(Access):
     @classmethod
     def _get_impl(cls):
         return cls.impl
+
+    @classmethod
+    def delete_by_query(cls, **query):
```

Comment:
This is done so we have control over which models should support purge. It's a little confusing for me when I looked at the code.